clients/node/client-rds-data-node/protocol/AwsRestJson1_1.ts

```diff
@@ -0,0 +1,214 @@
+import {
+  ColumnMetadata,
+  ExecuteStatementRequest,
+  ExecuteStatementResponse,
+  Field,
+  SqlParameter
+} from "../models/com/amazon/rdsdataservice";
+import { HttpRequest, HttpResponse } from "@aws-sdk/protocol-http";
+
+export function executeStatementAwsRestJson1_1Serialize(
+  input: ExecuteStatementRequest
+): HttpRequest {
+  let body: any = {};
+  if (input.resourceArn !== undefined) {
+    body.resourceArn = input.resourceArn;
+  }
+
+  if (input.secretArn !== undefined) {
+    body.secretArn = input.secretArn;
+  }
+
+  if (input.sql !== undefined) {
+    body.sql = input.sql;
+  }
+
+  if (input.database !== undefined) {
+    body.database = input.database;
+  }
+
+  if (input.schema !== undefined) {
+    body.schema = input.schema;
+  }
+
+  if (input.parameters !== undefined) {
+    body.parameters = sqlParameterListAwsRestJson1_1Serialize(input.parameters);
+  }
+
+  if (input.transactionId !== undefined) {
+    body.transactionId = input.transactionId;
+  }
+
+  if (input.includeResultMetadata !== undefined) {
+    body.includeResultMetadata = input.includeResultMetadata;
+  }
+
+  if (input.continueAfterTimeout !== undefined) {
+    body.continueAfterTimeout = input.continueAfterTimeout;
+  }
+
+  return new HttpRequest({
+    body: JSON.stringify(body),
+    path: "/execute",
+    method: "POST",
+    protocol: "https:",
+    headers: {
+      "Content-Type": "application/json"
+    }
+  });
+}
+
+export function executeStatementAwsRestJson1_1Deserialize(
+  output: HttpResponse
+): ExecuteStatementResponse {
+  let data: any = {};
+  if (output.body) {
+    data = JSON.parse(output.body);
```

Comment:
Here the `output.body` is not a binary data. It's actually `Readable` in node and `Blob` in browser. You should call streamcollector transfer them into binary. Example can be found here: https://github.com/aws/aws-sdk-js-v3/blob/226e54407c4979be935d44fb3afdc32fb4154683/packages/protocol-rest/src/RestParser.ts#L215-L234, You may need to import a runtime-specific stream collector like [node collector](https://github.com/aws/aws-sdk-js-v3/tree/master/packages/stream-collector-node) and [browser collector](https://github.com/aws/aws-sdk-js-v3/tree/master/packages/stream-collector-browser)